jax/_src/util.py

```diff
@@ -227,7 +227,15 @@ def weakref_lru_cache(call: Callable, maxsize=2048):
   lock = threading.Lock()
```

Comment:
Can you add a short class docstring? What exactly is weak in this cache? (Presumably the key, but... say so.)